src/adapters/dweb-staking.adapter.ts

```diff
@@ -0,0 +1,50 @@
+import {BindingScope, extensionFor, injectable} from '@loopback/core';
+import {BigNumber} from 'ethers';
+import {STAKING_ADAPTERS_EXTENSION_POINT} from '../keys.js';
+import {BaseStakingContractAdapter} from '../staking.js';
+// Use the full path to import instead of `../types`
+import {DwebStaking__factory} from '../types/factories/DwebStaking__factory.js';
+
+abstract class BaseDwebStakingContractAdapter extends BaseStakingContractAdapter {
+  // Any assets is supported
+  supportedAssets = [];
+
+  /**
+   * Get staked token ids for the given owner
+   * @param owner - Owner address
+   * @returns
+   */
+  async getStakedTokenIds(owner: string): Promise<BigNumber[]> {
+    const contract = DwebStaking__factory.connect(
+      this.contractAddress,
+      this.provider,
+    );
+    try {
+      return [(await contract.getStakeDetails(owner)).initialDeposit];
```

Comment:
Is the staking close to ERC20 or ERC721?

Reply:
From the code itself, it seems like ERC20 balance. In that case, we should implement `getStakedTokenBalance()`